Issue #338 documentation site

hmble opened the pull request:
This documentation site is made using [docsify](https://docsify.js.org) for use with Github Pages as we already have documentaion in `docs` folder. 

Preivew https://hmble.github.io/cash/#/

Let me know about your opinion.

hmble commented:
There are few UI improvement left. I will do once I get your input on this.

fabiospampinato commented:
It seems like a pretty good starting point, thanks!

After having browsed it a bit I would say:

- The default theme should be light and more inline with GitHub's, perhaps with a switch for switching to a dark theme or with automatic light/dark theme selection.
- Each available method should be available in the sidebar too, properly categorized.
- There should be a search bar that at least searches for the available methods.

Other than that it seems like it could work.

hmble commented:
Removed dark theme as I cannot add dark mode toggle to docs website
because of issues with current available plugings of docsify.

I will try to make new PR related to adding dark mode toggle if its
necessary. Until we can get docs site up and running with default light
theme.

chocolateboy commented on docs/_sidebar.md:
```diff
@@ -0,0 +1,6 @@
+* [Cash Method](cash.md)
+* [Migration Guide](migration_guide.md)
+* [Extending Cash](extending_cash.md)
+* [Partail Build](partial_builds.md)
```
\* Partail -> Partial

chocolateboy commented on docs/cash.md:
```diff
@@ -0,0 +1,89 @@
+# Cash Method
```
Plural would make more sense since there's more than one: Cash **Methods**

chocolateboy commented on docs/collection.md:
```diff
@@ -0,0 +1,767 @@
+# Collection Method
```
Plural would make more sense since there's more than one: Collection **Methods**

hmble commented:
@chocolateboy  Thanks for taking your time to check PR.

fabiospampinato commented:
@hmble I had a tab in my browser open about this PR, it doesn't seem like the issues I mentioned [here](https://github.com/fabiospampinato/cash/pull/345#issuecomment-638978773) have been addressed yet, when they are feel free to ping me.

hmble commented:
I was waiting for your input. I addressed your mentioned issues.
1. Made light mode default. ( I cannot add toggle for dark mode becuause of issue with docsify )
2. I added search bar
3. Properly categoriesed docs

fabiospampinato commented:
@hmble Personally I'd make the following changes:

- No custom scrollbars
- Better sidebar categorization
  - stuff like the migration guide, extending cash and partial builds can be moved into a shared parent, and showing subsections for each of them seem to only add to the noise to me 
  - stuff like collection methods should be grouped by category
  - possibly the sidebar should be expanded by default
  - naming methods like `fn.add ()` rather than `add` seems just noisier to me
  - it's weird that there's a "docs" section, what's the rest then?
- it's weird that I can't collapse/expand a section without opening it
- it's weird that if I click on "Cash Method" the document moves to the very top (so far so good), but then to get back to that section I need to click "Cash Method" again which collapses the sidebar section too for some reason
- it's weird that if I scroll search results far enough I get back to the sidebar sections
- if I search for "to" that query must be searched in methods' names first, like the results should contain all the methods that have "to" in their name, not stuff like "...main selec_to_r" method for Cash..."
- the "js" label in code blocks is useless
- each method should probably also have a link to the file that implements it on github
- each method should probably also have a link to the more in depth documentation from jQuery
- most importantly the entire thing must be generated from the readme or vice versa
- there probably are other changes that I'd make, like the font size looks too high to me, but for now lets address that I've mentioned

hmble commented:
Sorry, I can't work on this PR. Feel free to close.

fabiospampinato commented:
Sorry for the confusion, my idea was to get something much easier to navigate than the readme (multiple pages, search etc.) and potentially a bit more useful (linking to jQuery's docs etc.).

Thanks anyway!
